pysph/tools/ipython.py

```diff
@@ -0,0 +1,590 @@
+                min=1,
+                max=100,
+                step=1,
+                value=25,
+                description='vector width',
+                layout=widgets.Layout(width='300px'),
+            )
+            self.particles[particles_type].vector_width.owner = particles_type
+
+            self.particles[particles_type].vector_width.observe(
+                handlers['vector_width_handler'],
+                'value',
+            )
+
+            self.particles[particles_type].vector_scale = widgets.FloatSlider(
+                min=1,
+                max=100,
+                step=1,
+                value=55,
+                description='vector scale',
+                layout=widgets.Layout(width='300px'),
+            )
+            self.particles[particles_type].vector_scale.owner = particles_type
+            self.particles[particles_type].vector_scale.observe(
+                handlers['vector_scale_handler'],
+                'value',
+            )
+
+            self.particles[particles_type].scalar_size = widgets.FloatSlider(
+                min=0,
+                max=50,
+                step=1,
+                value=10,
+                description='scalar size',
+                layout=widgets.Layout(width='300px'),
+            )
+            self.particles[particles_type].scalar_size.owner = particles_type
+            self.particles[particles_type].scalar_size.observe(
+                handlers['scalar_size_handler'],
+                'value',
+            )
+
+
+class Viewer2D(Viewer):
+
+    '''
+    Example
+    -------
+
+    >>> from pysph.solver.utils import ipython
+    >>> sample = ipython.Viewer2D('/home/uname/pysph_files/dam_Break_2d_output')
+    >>> sample.interactive_plot()
+    >>> sample.show_log()
+    >>> sample.show_info()
+    '''
+
+    def _create_widgets(self):
+
+        handlers = {}
+        handlers['frame'] = self._frame_handler
+        handlers['scalar_handler'] = self._scalar_handler
+        handlers['vector_handler'] = self._vector_handler
+        handlers['vector_scale_handler'] = self._vector_scale_handler
+        handlers['vector_width_handler'] = self._vector_width_handler
+        handlers['scalar_size_handler'] = self._scalar_size_handler
+        handlers['save_figure'] = self._save_figure_handler
+        handlers['legend_handler'] = self._legend_handler
+
+        self._widgets = Viewer2DWidgets(
+            file=self.paths_list[0],
+            file_count=len(self.paths_list) - 1,
+            handlers=handlers
+        )
+
+    def _display_widgets(self):
+        '''
+        Display the widgets created using _create_widgets() method.
+        '''
+
+        from ipywidgets import HBox, VBox, Label, Layout
+
+        items = []
+        for particles_type in self._widgets.particles.keys():
+            items.append(
+                VBox([
+                    Label(particles_type),
+                    self._widgets.particles[particles_type].scalar,
```

Comment:
You seem to do this every time.  Instead of referring to `self._widgets.particles[particles_type].scalar`, reference this once above like so:
```
pa_widgets = self._widgets.particles['particles_types`]
...
VBox([
...
pa_widgets.scalar,
pa_widgets.vector,
...
]
...
)
```
However, in this case you can clean this up further as it smells that you create a VBox out of the widgets that belong to the widgets in this class.  Instead push the vbox creation also to the `ParticleArrayWidgets` into a method that simply returns a suitable VBox.